pkg/controllers/subnet/subnetbinding_handler.go

```diff
@@ -0,0 +1,103 @@
+package subnet
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+)
+
+func requeueSubnetBySubnetBindingCreate(ctx context.Context, c client.Client, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	newBM := objNew.(*v1alpha1.SubnetConnectionBindingMap)
+	enqueueSubnets(ctx, c, newBM, q)
+}
+
+func requeueSubnetBySubnetBindingUpdate(ctx context.Context, c client.Client, objOld client.Object, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	newBM := objNew.(*v1alpha1.SubnetConnectionBindingMap)
+	oldBM := objOld.(*v1alpha1.SubnetConnectionBindingMap)
+
+	if newBM.Spec.TargetSubnetName == oldBM.Spec.TargetSubnetName {
+		return
+	}
+
+	if newBM.Spec.TargetSubnetName != "" {
+		if err := enqueue(ctx, c, newBM.Namespace, newBM.Spec.TargetSubnetName, q); err != nil {
+			return
```

Comment:
same here, if failed to enqueue new targetSubnet, seems the old targetSubnet does not have chance to enqueue?

Reply:
updated.